### PR TITLE
Update evidenceHandled flag when attaching ER to an existing case

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-682

### Change description ###
Set the `evidenceHandled` flag value to `No` when attaching exception record to an existing case.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by aliveni** [comment]
> We need to set `evidenceHandled` to false on the case, not ExceptionRecord

@rkondratowicz I have made the implementation changes and functional test changes correct but integration test is wrong. I'll update the integration test. Please let me know if you think the other changes are not correct.

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/Documents.java [hunk 1]
```diff
@@ -76,13 +75,12 @@ private static String getDocumentId(Map<String, Object> document) {
     }
 
 
-    static Map<String, Object> insertNewRecords(List<Map<String, Object>> exceptionDocuments,
-                                                List<Map<String, Object>> existingDocuments) {
-        ImmutableList<Object> documentList = ImmutableList.builder()
+    static List<Object> addNewRecords(List<Map<String, Object>> exceptionDocuments,
+                                               List<Map<String, Object>> existingDocuments) {
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/Documents.java]: what we've ended up with now is exactly why it's better when there's a new line after `(` ;)

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/Documents.java [hunk 2]
```diff
@@ -76,13 +75,12 @@ private static String getDocumentId(Map<String, Object> document) {
     }
 
 
-    static Map<String, Object> insertNewRecords(List<Map<String, Object>> exceptionDocuments,
-                                                List<Map<String, Object>> existingDocuments) {
-        ImmutableList<Object> documentList = ImmutableList.builder()
+    static List<Object> addNewRecords(List<Map<String, Object>> exceptionDocuments,
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/Documents.java]: perhaps `concatDocuments` or `joinDocuments` would be clearer.

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java [hunk 3]
```diff
@@ -221,14 +223,20 @@ private void attachCaseByCcdId(
...
+            buildCaseData(exceptionRecordDocuments, targetCaseDocuments),
             createEventSummary(theCase, exceptionRecordId, exceptionRecordDocuments),
             event
         );
 
         log.info("Attached exception record '{}' to case with CCD ID '{}'", exceptionRecordId, targetCaseCcdRef);
     }
 
+    private Map<String, Object> buildCaseData(List<Map<String, Object>> exceptionDocuments,
+                                              List<Map<String, Object>> existingDocuments) {
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java]: let's use more often used (on this project at least) format:
```
functionName(
    Type1 param1,
    Type2 param2,
) {
```

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/Documents.java (hunk 2 above)
Comment (aliveni) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/Documents.java]: renamed the method

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java (hunk 3 above)
Comment (aliveni) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java]: changed format

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/Documents.java (hunk 1 above)
Comment (aliveni) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/Documents.java]: changed the method format

**Pull request merged**